packages/nimble-components/src/table-column/date-text/tests/table-column-date-text.spec.ts

```diff
@@ -26,187 +26,536 @@ async function setup(): Promise<Fixture<Table<SimpleTableRecord>>> {
+            await waitForUpdatesAsync();
+            expect(pageObject.getRenderedCellContent(0, 0)).toBe('10 PM');
+        });
+
+        it('honors customHourCycle property', async () => {
+            await element.setData([
+                { field: new Date('Dec 10, 2012, 10:35:05 PM').valueOf() }
+            ]);
+            await waitForUpdatesAsync();
+            column.format = 'custom';
+            column.customHour = 'numeric'; // must specify hour
+            column.customHourCycle = 'h24';
+            await waitForUpdatesAsync();
+            expect(pageObject.getRenderedCellContent(0, 0)).toBe('22');
+        });
+
+        it('honors customTimeZone and customTimeZoneName properties', async () => {
+            await element.setData([
+                { field: new Date('Dec 10, 2012, 10:35:05 PM UTC').valueOf() }
+            ]);
+            await waitForUpdatesAsync();
+            column.format = 'custom';
+            column.customTimeZone = 'UTC';
+            column.customTimeZoneName = 'short';
+            await waitForUpdatesAsync();
+            expect(pageObject.getRenderedCellContent(0, 0)).toBe(
+                '12/10/2012, UTC'
+            );
+        });
+
+        it('honors customDayPeriod property', async () => {
+            await element.setData([
+                { field: new Date('Dec 10, 2012, 10:35:05 PM').valueOf() }
+            ]);
+            await waitForUpdatesAsync();
+            column.format = 'custom';
+            column.customHour = 'numeric'; // must specify hour
+            column.customDayPeriod = 'narrow';
+            await waitForUpdatesAsync();
+            expect(pageObject.getRenderedCellContent(0, 0)).toBe('10 at night');
+        });
+
+        it('honors customCalendar property', async () => {
+            await element.setData([
+                { field: new Date('Dec 10, 2012, 10:35:05 PM').valueOf() }
+            ]);
+            await waitForUpdatesAsync();
+            column.format = 'custom';
+            column.customDay = 'numeric';
+            column.customMonth = 'short';
+            column.customYear = 'numeric';
+            column.customCalendar = 'hebrew';
+            await waitForUpdatesAsync();
+            expect(pageObject.getRenderedCellContent(0, 0)).toBe(
+                '26 Kislev 5773'
+            );
+        });
+
+        it('honors customNumberingSystem property', async () => {
+            await element.setData([
+                { field: new Date('Dec 10, 2012, 10:35:05 PM').valueOf() }
+            ]);
+            await waitForUpdatesAsync();
+            column.format = 'custom';
+            column.customYear = 'numeric';
+            column.customNumberingSystem = 'fullwide';
+            await waitForUpdatesAsync();
+            expect(pageObject.getRenderedCellContent(0, 0)).toBe('２０１２');
+        });
+
+        it('sets invalid flag on column when custom options are incompatible', async () => {
```

Comment:
Would it be valuable to add a few more tests around the validity of this columns? A few cases I can think of are:
- The column becomes valid again when custom options become compatible
- The column is initially valid and/or valid when not using custom formatting

Reply:
Added one test for each bullet